emails/tests/models_tests.py

```diff
@@ -475,242 +469,93 @@ def test_total_masks(self):
             baker.make(DomainAddress, user=self.profile.user, address=f"mask{i}")
         assert self.profile.total_masks == num_relay_addresses + num_domain_addresses
 
-    def test_at_mask_limit_premium_user_returns_False(self):
-        premium_user = baker.make(User)
-        baker.make(
-            SocialAccount,
-            user=premium_user,
-            provider="fxa",
-            extra_data={"subscriptions": [unlimited_subscription()]},
-        )
-        assert premium_user.profile.at_mask_limit is False
 
-    def test_at_mask_limit_free_user(self):
+class ProfileAtMaskLimitTest(ProfileTestCase):
+    """Tests for Profile.at_mask_limit"""
+
+    def test_premium_user_returns_False(self) -> None:
+        self.upgrade_to_premium()
         assert self.profile.at_mask_limit is False
-        for _ in list(range(settings.MAX_NUM_FREE_ALIASES)):
+
+    def test_free_user(self) -> None:
+        assert self.profile.at_mask_limit is False
+        for _ in range(settings.MAX_NUM_FREE_ALIASES):
             baker.make(RelayAddress, user=self.profile.user)
```

Comment:
suggestion (non-blocking): since we're in here and I just learned about `baker.make(_quantity=)`, maybe we can change this from a for loop?
```suggestion
        baker.make(
            RelayAddress,
            user=self.profile.user,
            _quantity=settings.MAX_NUM_FREE_ALIASES
        )
```